src/storage-controller/src/instance.rs

```diff
@@ -82,6 +88,15 @@ pub(crate) struct Instance<T> {
     response_tx: mpsc::UnboundedSender<(Option<ReplicaId>, StorageResponse<T>)>,
 }
 
+#[derive(Debug)]
+struct ActiveIngestion {
+    /// Whether the ingestion prefers running on a single replica.
+    prefers_single_replica: bool,
```

Comment:
Is this a property of an active ingestion? I would expect this information to be found in the collection state for the given `GlobalId`. Having it here implies that it can potentially disagree with the ingestion, right?

Reply:
It's a property of the `IngestionDescription`, and yeah, this is a cache of that, so that I don't have to re-derive it from the history. If you prefer that, I can also change it so we get it from the history instead. I don't think this is performance-sensitive code.